src/main/modules/nunjucks/getCheckAnswersRows.ts

```diff
@@ -0,0 +1,117 @@
+import { Case, Checkbox } from '../../app/case/case';
+import { TranslationFn } from '../../app/controller/GetController';
+import type { FormContent, FormOptions } from '../../app/form/Form';
+import { Sections, Step } from '../../steps/sequence';
+
+export const getCheckAnswersRows = function (section: Sections): GovUkNunjucksSummary[] {
+  const {
+    language,
+    isDivorce,
+    partner,
+    formState,
+    steps,
+  }: {
+    language: 'en' | 'cy';
+    isDivorce: boolean;
+    partner: string;
+    formState: Partial<Case>;
+    steps: ({ generateContent: TranslationFn; form: FormContent } & Step)[];
+  } = this.ctx;
+
+  return steps
+    .filter(step => step.showInSection === section)
+    .flatMap(step => {
```

Comment:
80 lines is just too long for an inline function. Split it into parts give them a name.

Also move some if not all of this logic into the app section. The modules stuff is really just for middleware set up.

Reply:
👍 I've refactored this into seperate method and files in 0f6aed0